sql/postgres/sqlspec.go

```diff
@@ -127,7 +128,70 @@ var (
 // ForeignKeySpecs into ForeignKeys, as the target tables do not necessarily exist in the schema
 // at this point. Instead, the linking is done by the convertSchema function.
 func convertTable(spec *sqlspec.Table, parent *schema.Schema) (*schema.Table, error) {
-	return specutil.Table(spec, parent, convertColumn, specutil.PrimaryKey, convertIndex, specutil.Check)
+	t, err := specutil.Table(spec, parent, convertColumn, specutil.PrimaryKey, convertIndex, specutil.Check)
+	if err != nil {
+		return nil, err
+	}
+	if err := convertPartition(spec.Extra, t); err != nil {
+		return nil, err
+	}
+	return t, nil
+}
+
+// convertPartition converts and appends the partition block into the table attributes if exists.
+func convertPartition(spec schemaspec.Resource, table *schema.Table) error {
+	r, ok := spec.Resource("partition")
+	if !ok {
+		return nil
+	}
+	var p struct {
+		Type    string            `spec:"type"`
+		Columns []*schemaspec.Ref `spec:"columns"`
+		Parts   []*struct {
+			Expr   string          `spec:"expr"`
+			Column *schemaspec.Ref `spec:"column"`
+		} `spec:"by"`
+	}
```

Comment:
you can do it in a subsequent PR but since we will want to marshal this to HCL the type should be defined outside the package

Reply:
👍